static/app/utils/performance/quickTrace/quickTraceQuery.tsx

```diff
@@ -63,7 +69,16 @@ export default function QuickTraceQuery({children, event, ...props}: QueryProps)
               traceFullResults.error === null &&
               traceFullResults.traces !== null
             ) {
-              for (const subtrace of traceFullResults.traces) {
+              if (
+                organization.features.includes(
+                  'performance-tracing-without-performance'
+                ) &&
+                isTraceFullSplitResult(traceFullResults.traces)
+              ) {
+                traceFullResults.traces = traceFullResults.traces.transactions;
+              }
+
+              for (const subtrace of traceFullResults.traces as TraceFull[]) {
```

Comment:
Hmmm, since the type of `traceFullResults.traces `is `TraceFull[] | | TraceFullSplitResults` to begin with it later complains since it can't really tell if `traceFullResults.traces` is an iterable.  
<img width="821" alt="Screenshot 2023-08-10 at 9 44 49 PM" src="https://github.com/getsentry/sentry/assets/60121741/284e5d19-e059-4acf-99d2-958830d1945f">

Reply:
Maybe we just wrap the loop with an isArray, removing the cast? @billyvg 
```
              if(Array.isArray(traceFullResults.traces)) {
                for (const subtrace of traceFullResults.traces) {
                  try {
                    const trace = flattenRelevantPaths(event, subtrace);
                    return children({
                      ...traceFullResults,
                      trace,
                      currentEvent: trace.find(e => isCurrentEvent(e, event)) ?? null,
                    });
                  } catch {
                    // let this fall through and check the next subtrace
                    // or use the trace lite results
                  }
                }
              }
```